rust/lance/src/io/writer.rs

```diff
@@ -365,16 +486,64 @@ impl FileWriter {
         }
 
         let positions: &dyn Array = &pos_builder.finish();
-        self.write_fixed_stride_array(field, &[positions]).await?;
+        Self::write_fixed_stride_array(object_writer, field, &[positions], batch_id, page_table)
+            .await?;
         let arrs = list_arrs.iter().collect::<Vec<_>>();
-        self.write_array(&field.children[0], arrs.as_slice()).await
+        Self::write_array(
+            object_writer,
+            &field.children[0],
+            arrs.as_slice(),
+            batch_id,
+            page_table,
+        )
+        .await
+    }
+
+    // For testing purposes, set statistics
+    // TODO: remove this once we have stats collection logic.
+    #[allow(dead_code)]
+    fn set_statistics(&mut self, stats: RecordBatch) {
+        self.stats = Some(stats);
+    }
+
+    async fn write_statistics(&mut self) -> Result<Option<StatisticsMetadata>> {
+        match &self.stats {
+            Some(stats_batch) if stats_batch.num_rows() > 0 => {
+                let schema = Schema::try_from(stats_batch.schema().as_ref())?;
+                let leaf_field_ids = schema.field_ids();
+
+                let mut stats_page_table = PageTable::default();
+                for (i, field) in schema.fields.iter().enumerate() {
+                    Self::write_array(
+                        &mut self.object_writer,
+                        field,
+                        &[stats_batch.column(i)],
+                        0, // Only one batch for statistics.
+                        &mut stats_page_table,
+                    )
+                    .await?;
+                }
+
+                let page_table_position = stats_page_table.write(&mut self.object_writer).await?;
+
+                Ok(Some(StatisticsMetadata {
+                    schema,
+                    leaf_field_ids,
+                    page_table_position,
+                }))
+            }
+            _ => Ok(None),
+        }
     }
 
     async fn write_footer(&mut self) -> Result<()> {
         // Step 1. Write page table.
         let pos = self.page_table.write(&mut self.object_writer).await?;
         self.metadata.page_table_position = pos;
 
+        // Step 1.5 Write statistics.
```

Comment:
Maybe we can just renumber the steps and call this step 2?

Reply:
Yeah I can reorder those